dev/python-detect-breaking-changes.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python3
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Original repository: https://github.com/StardustDL/aexpy
+# Copyright 2022 StardustDL <stardustdl@163.com>
+#
+
+# Usage: compare current code base with a commit or branch
+# python dev/python-detect-breaking-changes.py a540995345bc935db3b8ccf5cb94cb7caabc4847 (branch name or commit id)
```

Comment:
`python dev/python-detect-breaking-changes.py a540995345bc935db3b8ccf5cb94cb7caabc4847` works successfully,

while `python dev/python-detect-breaking-changes.py branch-3.4` fails with 
```
(spark_dev_310) ➜  spark git:(third-aexpy) python dev/python-detect-breaking-changes.py branch-3.4
Traceback (most recent call last):
  File "/Users/ruifeng.zheng/Dev/spark/dev/python-detect-breaking-changes.py", line 121, in diff
    with checkout(old, log) as src1:
  File "/Users/ruifeng.zheng/.dev/miniconda3/envs/spark_dev_310/lib/python3.10/contextlib.py", line 135, in __enter__
    return next(self.gen)
  File "/Users/ruifeng.zheng/Dev/spark/dev/python-detect-breaking-changes.py", line 85, in checkout
    runWithLog(
  File "/Users/ruifeng.zheng/Dev/spark/dev/python-detect-breaking-changes.py", line 63, in runWithLog
    result.check_returncode()
  File "/Users/ruifeng.zheng/.dev/miniconda3/envs/spark_dev_310/lib/python3.10/subprocess.py", line 457, in check_returncode
    raise CalledProcessError(self.returncode, self.args, self.stdout,
subprocess.CalledProcessError: Command '['git', 'worktree', 'add', '/Users/ruifeng.zheng/Dev/spark/dev/aexpy/cache/src/branch-3.4', 'branch-3.4']' returned non-zero exit status 128.
An error occured, view log at /Users/ruifeng.zheng/Dev/spark/dev/aexpy/cache/dist/__current__/with-branch-3.4/log.log .
(spark_dev_310) ➜  spark git:(third-aexpy)     
(spark_dev_310) ➜  spark git:(third-aexpy) cat /Users/ruifeng.zheng/Dev/spark/dev/aexpy/cache/dist/__current__/with-branch-3.4/log.log

subprocess.run(args=(['git', 'worktree', 'prune'],), kwargs={'cwd': PosixPath('/Users/ruifeng.zheng/Dev/spark')})




subprocess.run(args=(['git', 'worktree', 'add', '/Users/ruifeng.zheng/Dev/spark/dev/aexpy/cache/src/branch-3.4', 'branch-3.4'],), kwargs={'cwd': PosixPath('/Users/ruifeng.zheng/Dev/spark')})


fatal: invalid reference: branch-3.4

(spark_dev_310) ➜  spark git:(third-aexpy) 

```

Reply:
normally, the `dev` tools are used without `python` in the commands, like `./dev/python-detect-breaking-changes.py xxx`

Reply:
done, added the shebang `#!/usr/bin/env python3` to the script